app/models/relation.rb

```diff
@@ -207,6 +218,54 @@ def delete_with_history!(new_relation, user)
     end
   end
 
+  def self.delete_with_history_bulk!(relations, changeset, if_unused = false)
+    relation_hash = relations.collect { |r| [r.id, r] }.to_h
+    skipped = {}
+    # need to start the transaction here, so that the database can
+    # provide repeatable reads for the used-by checks. this means it
+    # shouldn't be possible to get race conditions.
+    Relation.transaction do
+      relation_ids = relation_hash.keys
+      old_relations = Relation.select("id, version, visible").where(:id => relation_ids).lock
+      raise OSM::APIBadUserInput, "Relation not exist. id: " + (relation_ids - old_relations.collect(&:id)).join(", ") unless relation_ids.length == old_relations.length
```

Comment:
Non-existing object should return HTTP 404 (Not found) instead of HTTP 400 (Bad Request)